From fe1b83b4131fd759dee644f1cb773b8b1721c31c Mon Sep 17 00:00:00 2001
From: Jacky Bai <ping.bai@nxp.com>
Date: Wed, 20 Jul 2022 12:33:25 +0800
Subject: [PATCH 251/274] LF-6624 soc: imx: Fix imx93 soc probe return failure
 error

Return directly from 'schedule_delayed_work' is not correct
in probe callback as its function type is 'bool', return true
when success, but probe is 'int', need to return '0' for success.
so refine the code to fix the probe failure issue.

Signed-off-by: Jacky Bai <ping.bai@nxp.com>
Reviewed-by: Ye Li <ye.li@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
---
 drivers/soc/imx/soc-imx9.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/imx/soc-imx9.c b/drivers/soc/imx/soc-imx9.c
index 0a3f4ba22..732d23349 100644
--- a/drivers/soc/imx/soc-imx9.c
+++ b/drivers/soc/imx/soc-imx9.c
@@ -121,7 +121,9 @@ static int imx9_init_soc_probe(struct platform_device *pdev)
 	 * A ELE ping request must be send at least once every day(24 hours),
 	 * so setup a delay work with 1 hour interval to ping sentinel periodically.
 	 */
-	return schedule_delayed_work(&ele_ping_work, ELE_PING_INTERVAL);
+	schedule_delayed_work(&ele_ping_work, ELE_PING_INTERVAL);
+
+	return 0;
 }
 
 static const struct of_device_id imx9_soc_of_match[] = {
-- 
2.25.1

